docs/reference/snapshot-restore/apis/get-snapshot-status-api.asciidoc

```diff
@@ -0,0 +1,273 @@
++
+To retrieve a list of all snapshots in the cluster, omit this and the `<snapshot>` parameter.
+
+`<snapshot>`::
+(Required, string)
+Comma-separated list of snapshot names to retrieve status for.
++
+To retrieve a list of all snapshots in a specified repository, omit this parameter.
++
+NOTE: Wildcard (`*`) expressions are not supported for `<snapshot>`.
+
+[role="child_attributes"]
+[[get-snapshot-status-api-request-body]]
+==== {api-request-body-title}
+
+`ignore_unavailable`::
+(Optional, boolean)
+If `false`, the request returns an error for any snapshots that are unavailable. Defaults to `false`.
++
+If `true`, the request ignores snapshots that are unavailable, such as those that are corrupted or temporarily cannot be returned.
+
+[role="child_attributes"]
+[[get-snapshot-status-api-response-body]]
+==== {api-response-body-title}
+`repository`::
+(string)
+Name of the repository that includes the snapshot.
+
+`snapshot`::
+(string)
+Name of the snapshot.
+
+`uuid`::
+(string)
+Universally unique identifier (UUID) of the snapshot.
+
+`state`::
++
+--
+(string)
+Indicates the current snapshot state.
+
+.Values for `state`
+[%collapsible%open]
+====
+`FAILED`::
+  The snapshot finished with an error and failed to store any data.
+
+`IN_PROGRESS`::
+  The snapshot is currently running.
+
+`PARTIAL`::
+  The global cluster state was stored, but data of at least one shard was not stored successfully.
+  The <<get-snapshot-api-response-failures,`failures`>> section of the response contains more detailed information about shards
+  that were not processed correctly.
+
+`SUCCESS`::
+  The snapshot finished and all shards were stored successfully.
+====
+--
+
+`include_global_state`::
+(boolean)
+Indicates whether the current cluster state is included in the snapshot.
+
+[[get-snapshot-status-shards-stats]]
+`shards_stats`::
+(object)
+Contains a count of shards in the snapshot.
++
+.Properties of `shards_stats`
+[%collapsible%open]
+====
+`initializing`::
+(integer)
+Number of shards that are still initializing.
+
+`started`::
+(integer)
+Number of shards that have started but not are not finalized.
+
+`finalizing`::
+(integer)
+Number of shards that are finalizing but are not done.
+
+`done`::
+(integer)
+Number of shards that initialized, started, and finalized successfully.
+
+`failed`::
+(integer)
+Number of shards that failed to be included in the snapshot.
+
+`total`::
+(integer)
+Total number of shards included in the snapshot.
+====
+
+[[get-snapshot-status-stats]]
+`stats`::
+(object)
+Provides details on the number (`file_count`) and size (`size_in_bytes`) of files included in the snapshot.
++
+.Properties of `stats`
+[%collapsible%open]
+====
+`incremental`::
+(object)
+Number and size of files that are not already in the repository and need to be copied as part of the incremental snapshot.
+
+`processed`::
+(object)
+Number and size of files that are still being copied.
```

Comment:
This is not correct. `processed` tracks those files that have already been uploaded. We increment the stats here after we upload a file

Reply:
Thanks for the correction. I'm proposing this text instead, which will also be reflected in [Restoring a snapshot](https://github.com/elastic/elasticsearch/blame/master/docs/reference/snapshot-restore/restore-snapshot.asciidoc#L258):

>Number and size of files that have already been uploaded to the snapshot. After a file is uploaded, the total `file_count` and `size_in_bytes` are incremented in `stats`.

Reply:
> After a file is uploaded, the total file_count and size_in_bytes are incremented in stats.

This is actually not correct. The total is set when the shard snapshot starts. It never changes during the snapshot. Both total and incremental counts are set at the beginning and the processed is then incremented as the snapshot executes.

Reply:
Ah, so the _processed_ `file_count` and `size_in_bytes` are incremented, not the _total.